numpy/_core/src/umath/matmul.c.src

```diff
@@ -499,9 +525,46 @@ NPY_NO_EXPORT void
                  * non-blasable (or non-ccontiguous output)
                  * we could still use BLAS, see gh-12365.
                  */
-                @TYPE@_matmul_inner_noblas(ip1, is1_m, is1_n,
-                                           ip2, is2_n, is2_p,
-                                           op, os_m, os_p, dm, dn, dp);
+                if(tmp_ip12op == NULL) {
+                    npy_intp ip1_size = sz * dm * dn,
+                             ip2_size = sz * dn * dp,
+                             op_size = sz * dm * dp,
+                             total_size = ip1_size + ip2_size + op_size;
+
+                    tmp_ip12op = (uint8_t*)malloc(total_size);
+
+                    tmp_ip1 = tmp_ip12op;
+                    tmp_ip2 = (uint8_t*)tmp_ip1 + ip1_size;
+                    tmp_op = (uint8_t*)tmp_ip2 + ip2_size;
+
+                }
+
+                /* Not enough memory */
+                if(tmp_ip12op == NULL) {
```

Comment:
```suggestion
                if (tmp_ip12op == NULL) {
```
Just nitpicking, but we put the space.

Reply:
I fixed all of those. Do we actually use a linter for the C/C++ code? Did not spot one yet. `cpplint` can supposedly do the trick with `cpplint --filter=-whitespace`

Reply:
There is a clang format file, that might work (but might also reformat everything else).  I would be happy to just apply a formatter to all the code at some point and maybe enforce it, but we never did...